sgx-tools/gen-quote.go

```diff
@@ -58,36 +52,17 @@ For example, generate the quote file according to the given local report file:
 			quotePath = "quote.bin"
 		}
 
-		rf, err := os.Open(reportPath)
-		if err != nil {
-			if os.IsNotExist(err) {
-				return fmt.Errorf("report file %s not found", reportPath)
-			}
-			return err
-		}
-		defer rf.Close()
-
-		var rfi os.FileInfo
-		rfi, err = rf.Stat()
+		report, err := readAndCheckFile(reportPath, intelsgx.ReportLength)
 		if err != nil {
 			return err
 		}
 
-		if rfi.Size() != intelsgx.ReportLength {
-			return fmt.Errorf("report file %s not match REPORT", reportPath)
-		}
-
-		buf := make([]byte, intelsgx.ReportLength)
-		if _, err = io.ReadFull(rf, buf); err != nil {
-			return fmt.Errorf("report file %s read failed", reportPath)
-		}
-
-		linkable := false
-		if context.Bool("linkable") {
-			linkable = true
+		quoteType := context.String("quoteType")
+		if !strings.EqualFold(quoteType, quoteTypeEcdsa) && !strings.EqualFold(quoteType, quoteTypeEpidUnlinkable) && !strings.EqualFold(quoteType, quoteTypeEpidLinkable) {
+			return fmt.Errorf("Unsupport quote type: %v", quoteType)
 		}
 
-		quote, err := intelsgx.GetQuote(buf, spid, linkable)
+		quote, err := intelsgx.GetQuoteEx(quoteType, report, context.String("spid"))
```

Comment:
spid in only valid for epid quote type. Plz add this check.

Reply:
> spid in only valid for epid quote type. Plz add this check.

Done, thanks, plz see line 65.

Reply:
spid has a fixed length. Plz add the length check.